### ISSUE TITLE
Tutoriel pour le mode TCP ?

### ISSUE THREAD

**Issue by Shunfry** [dorel14/3CX-Cdr-Server#63]
Bonjour,

Je suis sur une distri ubuntu 24, toute fraiche mais je n'arrive pas à finaliser la configuration du .env et lancer l'installation du docker.

Après avoir copier le .env_template, j'ai renseigné les infos mais je bloque au type de serveur.
J'aimerai faire du TCP avec le CDR actif comme évoqué dans le tutoriel, mais il n'y a aucun champs avec la mention #3CX server port comme énoncé ci-dessous.

![image](https://github.com/user-attachments/assets/3f108b67-0d4e-4cd8-bf0a-60de1995f56c)

Quand je fais le "docker compose build", j'ai une de ces deux erreurs :  
![image](https://github.com/user-attachments/assets/29639b29-7832-470b-8736-79537389bb11)
![image](https://github.com/user-attachments/assets/67c1b5ae-bdd5-416d-acf7-dbb0fae5ce25)

Donc, je ne sais pas ce qu'il ne va pas et le tutoriel dans "Wiki" ne précise rien dessus.

Peut-on m'aider dans la mise en service ?

En vous remerciant,
Cordialement.

**Comment by dorel14** [comment]
Bonjour 

Il y a un autre utilisateur qui m'a remonté le problème. J'étais en déplacement, je pense pouvoir avoir un correctif pour semaine prochaine

**Comment by SamKr** [comment]
Sorry my French is a bit rusty so I'll stick to English: your issues are fixed in the dev branch. The new .env example has the SERVER_PORT variable, and a bunch of explanations in the comments.

**Comment by Shunfry** [comment]
Don't worry I didn't expect a new collaborator on the project.
Thank you, I will try the fix.

**Comment by Shunfry** [comment]
Hello !
I got it working after get the "dev" branch, i could "docker compose build" then "docker compose up".
I could start every "webapp" but I got a problem with the "Server App" :

- First, in the tutorial, the url said that it's "localhost/app" but it not working after some checking I got that if I went to "localhost:8080", I got in the page.

![image](https://github.com/user-attachments/assets/1dcb5bfe-41fc-47eb-8ecc-f2a58a971fa4)

- second, tutorial said to go in the "3cx parameters" but either "extensions" or "queue" I have a error 500 :

![image](https://github.com/user-attachments/assets/5f72865c-6d39-456c-b235-b28c4ccbd909)

- Last, I configure my 3CX with active socket (port 5000), I used a public address and I create a NAT to redirect 5000 to my ubuntu.
How can I check if what I do is working ? Like see if the database grow ?
My PGadmin page :

![image](https://github.com/user-attachments/assets/441e1496-bd15-4fff-a0a5-a0652bc09576)

Thank you for your helps and your patience.
